### ISSUE TITLE
Models parsed via __response__ do not pass isinstance() checks against the original model

### ISSUE THREAD

**Issue by r4victor** [Ovsyanka83/pydantic-duality#13]
**Describe the bug**
We're parsing objects using `MyModel.__response__`, then pass the objects as if they were instances of `MyModel`. This fails because `isinstance(obj, MyModel)` return `False`.

**To Reproduce**
1. Define a model inherited from `DualBaseModel`
2. Parse an object using model's `__response__`.
3. Call `isinstance()`. The check fails:

```python
from pydantic_duality import DualBaseModel

class MyModel(DualBaseModel):
    one: str

my_model = MyModel.__response__.parse_obj({"one": "two", "extra": "three"})

print(isinstance(my_model, MyModel))
print(isinstance(my_model, MyModel.__response__))
# False
# True
```

**Expected behavior**
I'm not sure if this is intended or not and if it should be considered a bug in pydantic-duality, but such behavior was unexpected – it requires to always check for both `isinstance(obj, MyModel)` and `isinstance(obj, MyModel.__response__)` when the code works with objects parsed by `MyModel.__response__`.

**Screenshots**

**Operating system**
macOS

**Additional context**

**Comment by zmievsa** [comment]
Hm... I remember explicitly adding the feature to make sure this behavior is impossible. I'll take a look.

**Comment by zmievsa** [comment]
I looked into it. I did do this for reverse checks ( `isinstance(obj, cls.__request__)`) but not for the other types of checks. The current architecture is based on building completely different MRO for each object (where `cls.__request__` is based on the `.__request__` of its parents, and so on) which means that the classes you are trying to compare are completely unrelated. 

A completely different architecture seems to be necessary to fix the issue you are referencing. Do you have any ideas?

**Comment by zmievsa** [comment]
Fixed in 1.2.3. Please, check.
